cmd/zetacored/root.go

```diff
@@ -297,11 +377,20 @@ func (ac appCreator) newApp(
 		skipUpgradeHeights[int64(h)] = true
 	}
 
+	chainID, err := getChainIDFromOpts(appOpts)
+	if err != nil {
+		panic(err)
+	}
+
+	zetachain, err := chains.ZetaChainFromCosmosChainID(chainID)
+	if err != nil {
+		panic(err)
+	}
+
 	return app.New(logger, db, traceStore, true, skipUpgradeHeights,
 		cast.ToString(appOpts.Get(flags.FlagHome)),
 		cast.ToUint(appOpts.Get(server.FlagInvCheckPeriod)),
-		//cosmoscmd.EncodingConfig(ac.encCfg),
-		ac.encCfg,
+		uint64(zetachain.ChainId), //#nosec G115 chain id won't exceed uint64 // TODO evm: evm chain id?
```

Comment:
What does the TODO mean?

Reply:
evm chain id is hardcoded in config.go in cosmos evm and seems there is no way to make it configurable properly, this is what i need to see if it can be simplified somehow